src/storage/txn/actions/pessimistic_prewrite.rs

```diff
@@ -53,12 +54,19 @@ pub fn pessimistic_prewrite<S: Snapshot>(
             return Err(txn
                 .handle_non_pessimistic_lock_conflict(key, lock)
                 .unwrap_err());
-        } else {
-            if lock.lock_type != LockType::Pessimistic {
+        } else if lock.lock_type != LockType::Pessimistic {
+            // Allow to overwrite the primary lock to fallback from async commit.
+            if lock.use_async_commit && secondary_keys.is_none() {
+                info!("fallback from async commit"; "start_ts" => txn.start_ts);
+                fallback_from_async_commit = true;
```

Comment:
How about inheriting `TTL` from the lock?

Reply:
Is it necessary? Actually I think it's better to use the TTL in the new request. Fallback happens when one of the prewrite requests executes too slow and some time has passed, so it may be better for the new request to refresh TTL.

Reply:
From `twoPhaseCommitter`, it calculates the lockTTL at the beginning. If you can calculate a new one when fallback, it's OK.